moped-database/migrations/1711493493444_refactor_parent_audit_field_function/up.sql

```diff
@@ -0,0 +1,38 @@
+CREATE OR REPLACE FUNCTION update_audit_fields_with_dynamic_parent_table_name()
+RETURNS TRIGGER
+AS $$
+    DECLARE
+        parent_table_name text;
+        parent_table_primary_key_column_name text;
+        child_table_foreign_key_name text;
+        query_to_execute text;
+        project_id_variable INTEGER;
+BEGIN
+    parent_table_name = TG_ARGV[0];
+    parent_table_primary_key_column_name = TG_ARGV[1];
+    child_table_foreign_key_name = TG_ARGV[2];
+    
+    -- Build update to parent table
+    query_to_execute = format('
+        UPDATE %I SET updated_at = ''%s'', updated_by_user_id = %s WHERE %I.%I = $1.%I RETURNING project_id;', 
+        parent_table_name,
+        NEW.updated_at,
+        -- interpolate NULL as a string to handle when NEW.updated_by_user_id = NULL on CREATE
+        quote_nullable(NEW.updated_by_user_id),
```

Comment:
`quote_nullable` is really interesting thanks for discovering that Mike! i like the idea of using this little cheat code rather than patching the seed data 🤓

Reply:
This entire function is such a TIL for me, you mentioned that theres been leveling up of SQL lately, ✨ 

I agree with TIlly i like this "cheat code" and I'm wondering if future us would like the comment to point out that this only is needed with seed data. but maybe we will remember and thats not needed. im on the fence.